lib/ansible/modules/packaging/os/snap.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Stanislas Lange (angristan) <angristan@pm.me>
+# Copyright: (c) 2018, Victor Carceler <vcarceler@iespuigcastellar.xeill.net>
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: snap
+
+short_description: Manages snaps
+
+version_added: "2.8"
+
+description:
+    - "Manages snaps packages."
+
+options:
+    name:
+        description:
+            - Name of the snap to install or remove.
+        required: true
+    state:
+        description:
+            - Desired state of the package.
+        required: false
+        default: present
+        choices: [ absent, present ]
+    classic:
+        description:
+            - Confinement policy. The classic confinment allows a snap to have
+              the same level of access to the system as "classic" packages,
+              like those managed by APT. This option corresponds to the --classic argument.
+        type: bool
+        required: false
+        default: False
+
+author:
+    - Victor Carceler (vcarceler@iespuigcastellar.xeill.net)
+    - Stanislas Lange (angristan) <angristan@pm.me>
+'''
+
+EXAMPLES = '''
+# Install "foo" snap
+- name: Install foo
+  snap:
+    name: foo
+
+# Remove "foo" snap
+- name: Remove VLC
+  snap:
+    name: foo
+    state: absent
+
+# Install a snap with classic confinement
+- name: Install "foo" with option --classic
+  snap:
+    name: foo
+    classic: True
+'''
+
+RETURN = '''
+msg:
+    description: An informative message on the task's output
+    type: str
+    returned: on error
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def clean_err(err):
+    # Remove whitespaces and newlines that the snap CLI outputs.
+    return err.replace('\n', '').replace('      ', '').replace('"', '\'')
+
+
+def snap_exists(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s info %s" % (snap_path, snap_name)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc, out, err
+
+
+def is_snap_installed(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s list %s" % (snap_path, snap_name)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc, out, err
+
+
+def install_snap(module, snap_name):
+    rc, out, err = is_snap_installed(module, snap_name)
+    if rc == 0:
+        # Snap is already installed
+        return False
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    classic = '--classic' if module.params['classic'] else ''
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s install %s %s" % (snap_path, snap_name, classic)
```

Comment:
you could construct command as a list and then concatenate it right before calling. list is easier to work with.
it might look like:
```python
cmd_parts = [snap_path, 'install', snap_name]
if module.params['classic']:
    cmd_parts.append('--classic')

# and then
cmd = ' '.join(cmd_parts)
rc, out, err = module.run_command(cmd, check_rc=False)
```

Reply:
Awesome! Thanks for the tip